src/components/app.jsx

```diff
@@ -106,79 +32,89 @@ export default class App extends Component {
+				"click",
+				event => {
+					const target = event.target;
+					const showConsentToolButtonClicked = RegExp("showConsentTool").test(
+						target.getAttribute("onclick")
+					);
+					const appDiv = self.base;
+					const { layout } = config;
+					const {
+						isConsentToolShowing,
+						isFooterConsentToolShowing,
+						isThinConsentToolShowing,
+						isFooterShowing
+					} = store;
+					const isPopupVisible =
+						isConsentToolShowing ||
+						isFooterConsentToolShowing ||
+						isThinConsentToolShowing ||
+						isFooterShowing;
+
+					if (
+						isPopupVisible &&
+						!showConsentToolButtonClicked &&
+						!appDiv.contains(target)
+					) {
+						store.toggleConsentToolShowing(false);
+
+						// Render footer style CMP if no consent decision has been submitted yet
+						if (!cmp.submitted) {
+							layout !== "thin"
+								? store.toggleFooterConsentToolShowing(true)
+								: store.toggleThinConsentToolShowing(true);
+						}
 					}
-				};
-			}, false);
+				},
+				false
+			);
 		}
 	}
 
-	componentDidMount() {
-		const { store } = this.state;
-		const { config } = this.props;
-
-		if (config.css["custom-font-url"]) {
-			let head = document.head;
-			let link = document.createElement("link");
-			link.type = "text/css";
-			link.rel = "stylesheet";
-			link.href = config.css["custom-font-url"];
-			head.appendChild(link);
-		}
+	onFooterClose = () => {
+		this.props.store.toggleFooterShowing(false);
+	};
 
-		store.subscribe(this.updateCSSPrefs);
-		this.updateCSSPrefs();
-	}
+	onShowConsent = () => {
+		this.props.store.cmp.commands.showConsentTool();
+	};
 
 	render(props, state) {
 		const { store } = state;
 		const { config } = props;
-		const userLocalization = config.localization[currentLocale.split('-')[0]];
+		const showPopup =
+			store.isConsentToolShowing ||
+			store.isFooterConsentToolShowing ||
+			store.isThinConsentToolShowing;
+		const showFooter = store.isFooterShowing && config.showFooterAfterSubmit;
 
 		return (
-			<div class={style.gdpr}>
-				<Popup
-					store={store}
-					localization={userLocalization}
-					onSave={this.onSave}
-					config={config}
-					updateCSSPrefs={this.updateCSSPrefs}
-				/>
-				<PopupFooter
-					store={store}
-					localization={userLocalization}
-					onSave={this.onSave}
-					config={config}
-					updateCSSPrefs={this.updateCSSPrefs}
-				/>
-				<PopupThin
-					store={store}
-					localization={userLocalization}
-					onSave={this.onSave}
-					config={config}
-					updateCSSPrefs={this.updateCSSPrefs}
-				/>
-				<Footer
-					store={store}
-					localization={userLocalization}
-					config={config}
-					updateCSSPrefs={this.updateCSSPrefs}
-				/>
-			</div>
+			<LocalizationProvider
+				forceLocale={config.forceLocale}
+				language={store.consentLanguage}
+				translations={config.localization}
+			>
+				<ThemeProvider theme={config.theme || mapLegacyTheme(config.css)}>
+					<div class={style.gdpr}>
+						{showPopup && (
```

Comment:
@LassaLek The `showPopup` and `showFooter` booleans both evaluate to false, at least when running locally, and therefore these two components never render to the screen.